### PR TITLE
feat/thin-6: Adds Endpoints to Manage Device UUIDs

### PR DESCRIPTION
**Pull request by abdullahtariq91**
Adds endpoints to add/delete/retrieve device UUIDs
`POST /api/devices`
`GET /api/devices`
`GET /api/devices/1`
`DELETE /api/devices/1`

### CODE REVIEW

File: app/models/device.model.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import sequelize from '../../config/sequelize';
+import * as Sequelize from 'sequelize';
+
+const Device = sequelize.define('tb_client_devices', {
+  UUID: {
+    type: Sequelize.STRING(255),
+    allowNull: false
+  },
+  client_id: {
```
Comment (hamxabaig) [inline app/models/device.model.ts]: We need to add user_id there too, as wallet has `client_id` and `user_id`. After adding `user_id`, add the check to find by `user_id` in the controller methods.

File: app/interfaces/models.ts [hunk 2]
```diff
@@ -91,4 +91,12 @@ export interface IActivityLog {
   user_id: string;
   created_at?: Date;
   updated_at?: Date;
+}
+
+export interface IDevice {
+  id?: number;
+  client_id: string;
+  UUID: string;
```
Comment (hamxabaig) [inline app/interfaces/models.ts:99]: add user_id

File: app/modules/devices/devices.controller.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+import { IRequest, IResponse } from '../../interfaces/utils/IServer';
+import Device from '../../models/device.model';
+import to from 'await-to-js';
+import { IDevice } from '../../interfaces/models';
+
+export default class DevicesController implements IController {
+  public async post(req: IRequest, res: IResponse, next: restify.Next) {
+    try {
+      const deviceObj = req.body;
+      const [err, device] = <[Error, IDevice]> await to (Device.create({
```
Comment (hamxabaig) [inline app/modules/devices/devices.controller.ts]: You don't need to use `to` when you have only 1 await call inside a try catch.

**Review by hamxabaig — DISMISSED** [review]
@abdullahtariq91 Did you discuss this task with @mrmading. Otherwise looks so easy to me to add or remove a device that it doesn't fit the purpose of what we're trying to achieve.
Otherwise it looks good just to add `user_id` there.

**Comment by abdullahtariq91** [comment]
@hamxabaig User hits the endpoint after logging in and adds the device. These were the requirements: https://thinblock.slack.com/archives/CBAVCTDTQ/p1534406764000100

Yes, `user_id` should be added. Is it already retrieved in the middleware (like `client_id` is)? I think it would be helpful to do it there, since `user_id` will be used a lot in other places as well.

**Comment by hamxabaig** [comment]
@abdullahtariq91 `user_id` is not related to the auth thats why its not in the middleware, the client provides his own implementation of `user_id` because we're not storing/processing their users. So you need to add `user_id` in each endpoint in params/body.

File: app/models/device.model.ts (hunk 1 above)
Comment (abdullahtariq91) [reply app/models/device.model.ts]: You mean, a separate method/endpoint to retrieve devices by `user_id` provided, correct?

**Comment by hamxabaig** [comment]
@abdullahtariq91 you didn't add arguments to the mocked functions?

**Review by hamxabaig — CHANGES_REQUESTED** [review]
Missing arguments in mocked functions

**Comment by abdullahtariq91** [comment]
Added arguments. I think we'll need to fix up a lot of test cases previously done, later. 😀

**Comment by hamxabaig** [comment]
Yep, We need to fix them. I'll create tasks in JIRA to address them properly. Because i realised this that they don't break even if you change the params. Thats why i asked you to make them strict. Good work 👍

**Review by hamxabaig — APPROVED** [review]

**Comment by abdullahtariq91** [comment]
Yup, I was going by with how the previous ones were done, so kept missing arguments (especially in functions related to db).

**Pull request merged**